src/OpenTelemetry.Exporter.Zipkin/ZipkinExporterHelperExtensions.cs

```diff
@@ -43,8 +43,19 @@ public static TracerProviderBuilder AddZipkinExporter(this TracerProviderBuilder
             configure?.Invoke(exporterOptions);
             var zipkinExporter = new ZipkinExporter(exporterOptions);
 
-            // TODO: Pick Simple vs Batching based on ZipkinExporterOptions
-            return builder.AddProcessor(new BatchExportProcessor<Activity>(zipkinExporter));
+            if (exporterOptions.ExporterType == ZipkinExporterType.SimpleExportProcessor)
+            {
+                return builder.AddProcessor(new SimpleExportProcessor<Activity>(zipkinExporter));
```

Comment:
Do we need SimpleExportProcessor given zipkin is always using RPC?

Reply:
I think we can just leave it upto user to pick SimpleExportProcessor, if they really want. For example they suspect some issue in Batching, and want to try Simple export.
Or they want to see every span right away without any delay. (It can be achieved with batch with batchsize=1, as well i think).

Reply:
Has the Zipkin exporter been tested with the SimpleExportProcessor? I experimented with this in the past and it deadlocked. Here I believe https://github.com/open-telemetry/opentelemetry-dotnet/blob/b956a5f7e95d610d5f4f2db34bf17b7b9d802db1/src/OpenTelemetry.Exporter.Zipkin/ZipkinExporter.cs#L84

Assuming my memory serves me right and the potential for deadlock still exists, I think if we changed this to a `Task.Run(async...).GetAwaiter().GetResult()` it may avoid deadlock.

Reply:
@alanwest I haven't tested it, but I think it should be fine. The SimpleExportProcessor uses a lock to ensure only one thread is actually exporting at a time:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/67f77ab227fa2822fa12955b5b25ca28888e40d9/src/OpenTelemetry/SimpleExportProcessor.cs#L43

The original version would fire-and-forget a Task, which would be chaos 🐶 🐱 🏠

Reply:
You might be right. I can run a quick experiment again to verify.

Reply:
I tested both an ASP.NET and ASP.NET Core application and no deadlock occurred.

I'm always wary of blocking on asynchronous code invoked from synchronous code. Especially in the context of ASP.NET's SynchronizationContext as described [here](https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html).

Though, I think we're good 👍. The 🐶's and 🐱's are behaving.